drivers/serial/uart_b91.c

```diff
@@ -0,0 +1,568 @@
+/*
+ * Copyright (c) 2021 Telink Semiconductor
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "analog.h"
+#include "clock.h"
+
+#include <device.h>
+#include <drivers/uart.h>
+#include <drivers/pinmux.h>
+#include <dt-bindings/pinctrl/b91-pinctrl.h>
+
+
+/* Driver dts compatibility: telink,b91_uart */
+#define DT_DRV_COMPAT telink_b91_uart
+
+/* Get UART instance */
+#define GET_UART(dev)      ((volatile struct uart_b91_t *) \
+			    ((const struct uart_b91_config *)dev->config)->uart_addr)
+
+/* Get UART configuration */
+#define GET_CFG(dev)       ((const struct uart_b91_config *)dev->config)
+
+/* Get instance data */
+#define DEV_DATA(dev)      ((struct uart_b91_data *const)dev->data)
+
+/* UART TX buffer count max value */
+#define UART_TX_BUF_CNT    ((uint8_t)8u)
+
+/* Parity type */
+#define UART_PARITY_NONE   ((uint8_t)0u)
+#define UART_PARITY_EVEN   ((uint8_t)1u)
+#define UART_PARITY_ODD    ((uint8_t)2u)
+
+/* Stop bits length */
+#define UART_STOP_BIT_1    ((uint8_t)0u)
+#define UART_STOP_BIT_1P5  BIT(4)
+#define UART_STOP_BIT_2    BIT(5)
+
+
+/* B91 UART registers structure */
+struct uart_b91_t {
+	uint8_t data_buf[4];
+	uint8_t clk_div_l;
+	uint8_t clk_div_h;
```

Comment:
I assume B91 is little endian and there will never be a big endian variant. In that case, any reason we can't do `uint16_t clk_div_l` here? Do these peripherals have only lower 8 bits connected to the data bus?

Reply:
16 bits are connected.
Fixed.